nom-sql/src/column.rs

```diff
@@ -88,7 +88,7 @@ pub enum ColumnConstraint {
     Unique,
     /// NOTE(grfn): Yes, this really is its own special thing, not just an expression - see
     /// <https://dev.mysql.com/doc/refman/8.0/en/timestamp-initialization.html>
-    OnUpdateCurrentTimestamp,
+    OnUpdateCurrentTimestamp(Option<Expr>),
```

Comment:
it doesn't seem like MySQL accepts an arbitrary expression here, only an integer:

```sql
MySQL [none]> CREATE TABLE t1 (
    ->   ts TIMESTAMP(6) DEFAULT CURRENT_TIMESTAMP(6) ON UPDATE CURRENT_TIMESTAMP(5 + 1)
    -> );
ERROR 1064 (42000): You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '+ 1)
)' at line 2
```

Reply:
Agreed, however I was just copying the same pattern that is used for parsing `DEFAULT CURRENT_TIMESTAMP(6)` - which accepts `DEFAULT CURRENT_TIMESTAMP(5+1)`

Reply:
I've restricted the parser to accepting `literal` and optional whitespace and updated test case accordingly